server/accounts.go

```diff
@@ -2893,6 +2893,8 @@ func (dr *CacheDirAccResolver) Fetch(name string) (string, error) {
 	var err error
 	var theJWT string
 	select {
+	case <-s.quitCh:
```

Comment:
I just realized that `s.sys` is set to `nil` as part of the shutdown of the eventing. So we have to be extremely careful to check for `s.sys != nil`. Personally, I don't like that we set things to nil when they are possibly used in various go routines. I generally prefer to have a close state that is safer than dereferencing a nil pointer...
With current behavior, could you make sure that new code that you have added in this PR make sure that `s.sys` is not nil?

Reply:
while receiving from a nil channel is not a problem (https://play.golang.org/p/qmzBSkPlb3k) I am now reading it while holding the lock

Reply:
Sorry, I put the comment here, but I am talking about `s.sys`, not `s.quitCh`. So any place where you do `s.sys.replies`, etc..

Reply:
Fixed by checking for nil and then retaining the map. (avoids two more checks)